src/Cli/ConfigGenerator.cs

```diff
@@ -82,46 +87,46 @@ public static bool TryCreateRuntimeConfig(InitOptions options, out string runtim
                         return false;
                     }
 
-                    if (!File.Exists(graphQLSchemaPath))
+                    if (!fileSystem.File.Exists(graphQLSchemaPath))
                     {
                         _logger.LogError($"GraphQL Schema File: {graphQLSchemaPath} not found.");
                         return false;
                     }
 
                     // If the option --rest.path is specified for cosmosdb_nosql, log a warning because
                     // rest is not supported for cosmosdb_nosql yet.
-                    if (!GlobalSettings.REST_DEFAULT_PATH.Equals(restPath))
+                    if (!RestRuntimeOptions.DEFAULT_PATH.Equals(restPath))
                     {
-                        _logger.LogWarning("Configuration option --rest.path is not honored for cosmosdb_nosql since " +
-                            "it does not support REST yet.");
+                        _logger.LogWarning("Configuration option --rest.path is not honored for cosmosdb_nosql since it does not support REST yet.");
                     }
 
                     restPath = null;
-                    dbOptions = new CosmosDbNoSqlOptions(cosmosDatabase, cosmosContainer, graphQLSchemaPath, GraphQLSchema: null);
+                    dbOptions.Add(namingPolicy.ConvertName(nameof(CosmosDbNoSQLDataSourceOptions.Database)), JsonSerializer.SerializeToElement(cosmosDatabase));
+                    dbOptions.Add(namingPolicy.ConvertName(nameof(CosmosDbNoSQLDataSourceOptions.Container)), JsonSerializer.SerializeToElement(cosmosContainer));
+                    dbOptions.Add(namingPolicy.ConvertName(nameof(CosmosDbNoSQLDataSourceOptions.Schema)), JsonSerializer.SerializeToElement(graphQLSchemaPath));
                     break;
 
-                case DatabaseType.mssql:
-                    dbOptions = new MsSqlOptions(SetSessionContext: options.SetSessionContext);
+                case DatabaseType.MSSQL:
+                    dbOptions.Add(namingPolicy.ConvertName(nameof(MsSqlOptions.SetSessionContext)), JsonSerializer.SerializeToElement(options.SetSessionContext));
+
                     break;
-                case DatabaseType.mysql:
-                case DatabaseType.postgresql:
-                case DatabaseType.cosmosdb_postgresql:
+                case DatabaseType.MySQL:
+                case DatabaseType.PostgreSQL:
+                case DatabaseType.CosmosDB_PostgreSQL:
                     break;
                 default:
                     throw new Exception($"DatabaseType: ${dbType} not supported.Please provide a valid database-type.");
             }
 
-            DataSource dataSource = new(dbType, DbOptions: dbOptions);
+            DataSource dataSource = new(dbType, string.Empty, dbOptions);
```

Comment:
Changed the flow a little to read the `options.ConnectionString` inline rather than in a follow-up `if` statement. This makes it clearer what is being assigned and also removes a memory allocation that is discarded almost immediately.